lib/Connection.js

```diff
@@ -134,49 +166,46 @@ class Connection {
    * Invokes transport with XML input from command list
    * Calls user provided callback with the XML output
    * @param {fuction} callback
-   * @param {boolean} sync
    */
   run(callback) {
+    if (typeof callback !== 'function') {
+      throw new Error('Passing a callback function is required');
+    }
     // Build the XML document.
-    let xml = iXml.iXmlNodeHead() + iXml.iXmlNodeScriptOpen();
+    let xmlInput = iXmlNodeHead() + iXmlNodeScriptOpen();
+
     for (let i = 0; i < this.cmds.length; i += 1) {
-      xml += this.cmds[i];
+      xmlInput += this.cmds[i];
     }
-    xml += iXml.iXmlNodeScriptClose();
-    this.cmds = [];
+
+    xmlInput += iXmlNodeScriptClose();
```

Comment:
What about
```javascript
xmlInput = iXmlNodeHead() + iXmlNodeScriptOpen() + this.cmds.join() + iXmlNodeScriptClose();
```

Reply:
That should work and avoid the for loop.